xontrib/hist_navigator.py

```diff
@@ -90,24 +139,113 @@ def insert_text(event, text):
     carriage_return(b, event.cli)
 
 
+def _cd_inline(path: typing.Optional[typing.AnyStr] = None) -> None:
+    """Change dir without creating a new prompt line, updating existing instead"""
+
+    if path is None:
+        args = []
+    elif isinstance(path, bytes):
+        args = [path.decode("utf-8")]
+    elif isinstance(path, str):
+        args = [path]
+    if         path and \
+      not Path(path).is_dir():
+        return # do nothing is target is not a Dir
+    _, exc, _ = cdx(args)
+    if exc is not None:
+        raise Exception(exc)
+    else:
+        t = threading.Thread(target=_update_prompt, args=())
+        t.start()
+
+
+def _parse_key_user(key_user):
+    _key_symb = {
+        '⎈':'c-'  ,'⌃':'c-'   ,
```

Comment:
it is better to avoid new abstractions and use the default ptk's keymap. Also it uses some non-ascii chars.

Reply:
> it uses some non-ascii chars.

Sure, that's the beauty of it. Instead of the ugly ascii that blends together modifiers with keys you can use visually distinct config

For example, this is way more readable (notice how it's also easy to differentiate which modifiers are used)
<img width="80" alt="kb_new" src="https://user-images.githubusercontent.com/12956286/225325732-44e1f7b7-d578-477e-a086-8c2b773f5ab0.png">
vs this
<img width="158" alt="kb_old" src="https://user-images.githubusercontent.com/12956286/225325720-2139eb55-60d5-4c9e-81dd-51172c3c6557.png">


Also, why would you force users to type `["escape","s"]` ptk's keymap instead of `a-s` that this abstraction allows?

Reply:
Because they already familiar with ptk key handler functions. Also there exist documentation explaining the key names. It is also more readable

Reply:
> Because they already familiar with ptk key handler functions

So how does this stop them??? Also, this doesn't explain why you want to force the bad interface on the majority of users who aren't familiar (and it's not even mentioned anywhere that these vars are input to the handler functions). All of the benefits are __opt in__

Reply:
and you're able to use `envx["XSH_HISTNAV_KEY_PREV"]	= ['escape','h' ]` here

Reply:
But there is a need and a few benefits listed above that cover the need. What's the downside of having a better user interface?

Reply:
It feels obscured

Reply:
Don't get it, what does the simple text replacement obscure and from whom?

Reply:
Yeah, but you still can't explain what the issue is, and without any downside I don't see the point in degrading the user experience (especially when users who like don't need to switch)

Reply:
the issue is it is adding one more layer which is not needed. If you feel this is needed, then please create another PR which just maps the keys if you feel that improves user experience. I find it hard to type symbols from my keyboard.

Reply:
> I find it hard to type symbols from my keyboard

You don't need to type any symbols, we've already discussed this, symbols are **opt in**

Same with the need, I've explained what the need is, it's easier to type (even if you ignore the symbols, 'a-x' is better than ['escape','x']) and possible to vertically align for clarity

Anyway, we're going in circles, it's up to you what you want to incorporate into this xontrib, but I won't make it worse for no reason, especially when I'd anyway need to use the better fork (I use shorter aligned input in my configs, and it's great)

Reply:
the short forms already exists with ptk. I dont want to add any more custom mapping. I don't want the code to be complex.

Reply:
what's the short form for your default keybind Alt + Left Arrow in ptk?

Reply:
Yes Alt doesn't have a shortcut because of compatibility reasons. It emits the escape code.

Reply:
So short forms don't really exist with ptk for some important keys. And compatibility doesn't explain the bad user APIs as a simple string replacement can solve the issue

Reply:
`["escape", "left"]` is more readable than `"⎇←" `. I do not want a mapping of keys. Please separate that into a new PR if you want to work on that. Otherwise feel free to use a fork. This is fairly a small change.

Reply:
I don't know why you keep repeating the same trivial mistakes trying to defend a position you can't come up with a single argument for
1. It's not just `"⎇←"`, it's also `"a-left"` or could be `"alt-left"`
2. it makes no sense that for the alt key `escape` is more readable than `a-`/`alt-`
3. No one is forcing you to use `⎇`, `["escape", "left"]` is still available if you think it's readable

And it's not about the size of a change, it's about the fact that you won't to waste effort making the users worse off without a reason, so sure I'll use the more ergonomic fork instead

Reply:
good then. Please check the zen of Python.

Reply:
I have, you fail at the first principle

> Beautiful is better than ugly.

Reply:
the symbols look ugly to me.

Reply:
and you don't have to use them, `a-`/`alt-` still exists despite the fact that you keep ignoring it

Reply:
I believe I answered that it has compatiblity issues that is the reason no short form exists with ptk itself. This is not going anywhere feel free to fork and use it.

Reply:
And I believe I explained how your answer is wrong (not that it ever prevents repetition)

> compatibility doesn't explain the bad user APIs as a simple string replacement can solve